lib/github.ts

```diff
@@ -651,6 +659,43 @@ export class FakeGitHub implements GitHub {
     return result;
   }
 
+  public forkRepository(repository: string | GitHubRepository, forkedRepositoryOwner: string): Promise<FakeGitHubRepository> {
+    return this.getRepository(repository)
+      .then((fakeRepository: FakeGitHubRepository) => {
+        let result: Promise<FakeGitHubRepository>;
+
+        const repositoryName: GitHubRepository = getGitHubRepository(fakeRepository.name);
+        const forkedRepositoryFullName = `${forkedRepositoryOwner}/${repositoryName.name}`;
+        if (contains(fakeRepository.forks, (forkedRepository: FakeGitHubRepository) => forkedRepository.name === forkedRepositoryFullName)) {
+          result = Promise.reject(new Error(`A fake repository fork of "${fakeRepository.name}" already exists for user/organization "${forkedRepositoryOwner}".`));
+        } else {
+          const forkedRepository = new FakeGitHubRepository(`${forkedRepositoryOwner}/${repositoryName.name}`, fakeRepository);
+          this.repositories.push(forkedRepository);
+          fakeRepository.forks.push(forkedRepository);
+          result = Promise.resolve(forkedRepository);
+        }
+
+        return result;
+      });
+  }
+
+  public deleteRepository(repository: string | GitHubRepository): Promise<void> {
+    const repositoryFullName: string = getRepositoryFullName(repository);
+    const deletedRepository: FakeGitHubRepository | undefined = removeFirst(this.repositories, (repo: FakeGitHubRepository) => repo.name === repositoryFullName);
+
+    let result: Promise<void>;
+    if (!deletedRepository) {
+      result = Promise.reject(new Error(`No fake repository exists with the name "${repositoryFullName}".`));
+    } else {
+      if (deletedRepository.forkOf) {
+        removeFirst(deletedRepository.forkOf.forks, (fork: FakeGitHubRepository) => fork === deletedRepository);
+      }
+      result = Promise.resolve();
```

Comment:
If you return here and @688 you don't need additional variable

Reply:
Oh, I know. But I really don't like multiple return statements in a function, so the extra variable is worth it to me.